### DISCUSSION TITLE
Internet search if string not found

### DISCUSSION THREAD

**Discussion by jayrajput** [agzam/browser-hist.el#18] [Q&A] [answered]
Currently if I type on the browser address bar, it allows me to do these two
- Match against past URL and Title
- Allow a new search using the configured search engine.

How can I do a new search rather than matching against part URL and Title?

I am looking at ideas on workflow on how to achieve it. I already have a separate function which can do internet searches for me, but rather than switching between `browser-hist-search` and `internet-search`, I am trying to find how to unified my workflow similar to what is available in the browsers.

**Comment by agzam** [comment, marked as answer]
If I understand it right, you want to open the popup, start typing something, and if it finds something in the browser history - great, if not, you want to issue a google search?

What you are looking for can be done via using [Embark-become](https://github.com/oantolin/embark?tab=readme-ov-file#switching-to-a-different-command-without-losing-what-youve-typed). However, browser-hist.el doesn't out of the box provide easy way to hook into that, at least I don't know a trivial way to achieve it easily. IMO, for that to work we need to make the command Consult-compatible, which @karthink  and I discussed [here](https://github.com/agzam/browser-hist.el/issues/14) and [here](https://github.com/agzam/browser-hist.el/pull/15).

Basically, you'd call `browser-hist-search`, then start typing, then press `embark-become` key, that maybe allows you to pass the query to engine-mode command (for example). `engine-mode.el` is a package that allows you to define and use different query search engines - google, duckduckgo, wikipedia, etc.

There are some other benefits to make the command Consult-compatible. Hopefully one day I'll get some bandwidth and make that happen.

But if you're asking for a predictive search-like thing, that like Copilot tries to autocomplete URLs (that don't exist in the browser history) based on partial query as you type - that would be a completely different kind of conversation, and I'm not sure it would be even within the scope of this package.

**Reply by jayrajput** [reply]
`embark-become` is what I had in mind. Thanks for the feedback!! I will try myself based on your feedback.
